addons/point_of_sale/controllers/main.py

```diff
@@ -84,3 +84,64 @@ def print_sale_details(self, date_start=False, date_stop=False, **kw):
         pdf, _ = request.env['ir.actions.report'].with_context(date_start=date_start, date_stop=date_stop)._render_qweb_pdf('point_of_sale.sale_details_report', r)
         pdfhttpheaders = [('Content-Type', 'application/pdf'), ('Content-Length', len(pdf))]
         return request.make_response(pdf, headers=pdfhttpheaders)
+
+    @http.route(['/pos/ticket'], type='http', auth="user", website=True)
+    def show_ticket_form(self, order_reference='', ticket_amount='', error_message=''):
+        return request.render("point_of_sale.ticket_entry", {
+            'order_reference': order_reference,
+            'ticket_amount': ticket_amount,
+            'error_message': error_message,
+        })
+
+    @http.route(['/pos/ticket/validate'], type='http', auth="user", website=True)
+    def show_ticket_validation_screen(self, order_reference='', ticket_amount=''):
+        if not order_reference or not ticket_amount:  # The route is called directly
+            return request.redirect('/pos/ticket')
+        partner = request.env.user.partner_id
+        template_vals = {
+            'partner': partner,
+            'format_amount': format_amount,
+            'env': request.env,
+            'order_reference': order_reference,
+            'ticket_amount': ticket_amount,
+            'address_url': f'/my/account?redirect=/pos/ticket/validate?order_reference={order_reference}%26ticket_amount={ticket_amount}',
+        }
+        # Check that an order exists for the information entered
+        pos_order = request.env['pos.order'].sudo().search([('amount_total', '=', ticket_amount), ('pos_reference', '=like', f'Order {order_reference}')])
```

Comment:
'Order' is translatable, so it won't work in other languages than english, you should ptobably just check that the string ends with the reference.

https://github.com/odoo/odoo/blob/master/addons/point_of_sale/static/src/js/models.js#L2261